### ISSUE TITLE
New standard: Compression Streams

### ISSUE THREAD

**Issue by saschanaz** [whatwg/sg#231]
Following https://github.com/WICG/compression/issues/32, I propose standardizing Compression Streams API in WHATWG. It's an API that allows compressing and decompressing BufferSource types using a set of the most common compression algorithms (gzip/deflate/deflate-raw) in browsers through Stream API.

@ricea is the editor and I'm helping him for the transition.

`compression` sounds like a good short name, matching the current short name in WICG.

One open question that is blocking me from writing a full workstream proposal is that whether this matches in the existing Streams workstream. The current scope looks explicitly limited to low level stream API:

https://github.com/whatwg/sg/blob/ab9abf930e84ed83770ce4daa3a0aacd56c897de/db.json#L436

But @ricea is also an editor in Streams spec, so it may still make sense to have a common workstream?

**Comment by annevk** [comment]
Hey @saschanaz, thanks for raising this!

This is something we need the entire @whatwg/sg to chime in on. From my perspective it can reasonably be considered in scope of the Streams Workstream and thus not require a new one. Our next meeting is April 23, but I'll see if I can get everyone to comment here before then.

**Comment by annevk** [comment]
Apologies for the delay, the meeting got canceled as people are/have been traveling. The next meeting is scheduled for May 7. Hopefully people find some time to comment before then though as I don't really think there's anything controversial here.

**Comment by cwilso** [comment]
OK with me.

**Comment by yoavweiss** [comment]
Yay! Let me know when the old repo needs to be archived and/or have redirects put in place.

**Comment by annevk** [comment]
I think it would be ideal to move the repository wholesale into the WHATWG, via @domenic / @foolip / I as intermediary (I believe it cannot be transferred directly). Should probably wait until @saschanaz and @ricea have had a chance to go through the checklist.

**Comment by yoavweiss** [comment]
Yeah, that would work as well, but I believe we'd need someone with owner access to both orgs. Maybe @cwilso has that?

**Comment by foolip** [comment]
Transferring directly would be ideal for redirects. (I'm not sure if GitHub would simplify a A&rarr;B&rarr;C redirect to A&rarr;C.)

**Comment by annevk** [comment]
@yoavweiss the people I listed have that. @foolip that should work and is how we've done it before.

**Comment by saschanaz** [comment]
Alright, for https://github.com/whatwg/meta/blob/main/NEW-STANDARD.md, the shortname would be `compression` as it already is in WICG.

2 would be fun 😅 any idea what should it look like? I can only imagine WinZip-like icon, and adobe stock has something like this. 

![image](https://github.com/whatwg/sg/assets/3396686/59f9df74-cf7e-4025-bba5-5a651650db78)

**Comment by annevk** [comment]
https://www.svgrepo.com/svg/238546/compress is under CC0 and seems reasonable enough. Up to you though. @saschanaz are you willing to PR db.json as well?

**Comment by saschanaz** [comment]
Before having an actual repo transferred? If that's okay then yes.

**Comment by annevk** [comment]
Yeah that's fine. We won't merge until we have everything in order, but the more that's ready the easier it'll be to do the final administrative bits.

**Comment by foolip** [comment]
I now have transferred the repo using temporary WICG powers granted by @yoavweiss. I didn't give any teams access and haven't done anything else.

**Comment by annevk** [comment]
* I merged the logo.
* I updated some of the repository settings.
* I ran spec-factory and put up https://github.com/whatwg/compression/pull/57 as a result.
* Left a couple questions in https://github.com/whatwg/sg/pull/236 and suggested RD publication months (new months as all existing publication months already have 5 documents).

What needs to happen still:

- The spec-factory PR needs review.
- @foolip or @domenic need to setup the domain.
- Someone needs to update `index.bs` to account for the WHATWG template and update the IPR section to acknowledge where this effort started, similar to https://urlpattern.spec.whatwg.org/#ipr.
- Need to go through https://github.com/whatwg/meta/blob/main/NEW-STANDARD.md again to make sure all boxes are ticked. There's some more repository settings to fix at least.
